src/test/java/seedu/internbuddy/model/application/DescriptionTest.java

```diff
@@ -0,0 +1,52 @@
+package seedu.internbuddy.model.application;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+public class DescriptionTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        //assertThrows(NullPointerException.class, () -> new Description(null));
+    }
+
+    @Test
+    public void isValidDescription() {
+        // null description
+        assertThrows(NullPointerException.class, () -> Description.isValidDescription(null));
+
+        //invalid descriptions
+        assertFalse(Description.isValidDescription("")); // empty string
+        assertFalse(Description.isValidDescription(" ")); // spaces only
+
+        //valid descriptions
+        assertTrue(Description.isValidDescription("INTERESTED"));
+        assertTrue(Description.isValidDescription("APPLIED"));
+        assertTrue(Description.isValidDescription("CLOSED"));
+    }
+
+    @Test
+    public void equals() {
+        Description description1 = new Description("INTERESTED");
+        Description description2 = new Description("INTERESTED");
+        Description description3 = new Description("APPLIED");
```

Comment:
Minor nit: I think using different test strings (like: "DevOps Intern" or "Database Engineer", etc.) for the description to simulate a real application could be more appropriate.